### DISCUSSION TITLE
Example of proper setup without a storyboard

### DISCUSSION THREAD

**Discussion by mbechard** [KhronosGroup/MoltenVK#1586] [Q&A]
Is there any chance of an example of creating a window completely in code using NSWindow/ViewController etc, without using a storyboard? The reason I ask is because I have an easily reproducible bug where when resizing my swap chain due to a window resize, the swap chain can get into a state where it shows a corrupt image for every 2nd swap. The only debugging information I can get so far is a print from Xcode saying 'GPU Soft Fault count: 1'.
It would be great to get a gut-check example doing all-code setup to ensure my relationships between the view/layer/surface and swap chains are all correct.
Thanks!

**Comment by Zorro666** [comment]
I find the Metal-CPP project useful for that. https://developer.apple.com/metal/cpp/ the learn examples have window/view creation without any Objective C: https://developer.apple.com/metal/sample-code/ "Learn Metal with C++"

You are welcome to look at my own MoltenVk draw a triangle on screen project which does its own simplistic Window/View handling. https://github.com/Zorro666/VulkanTriangle

**Reply by mbechard** [reply]
Thanks, I'm actually not looking to avoid objective-c, just the storyboard. Thanks for your example as well, unfortunetely my bug occurs in particular when resizing the window though so your sample doesn't cover that test case it seems. Appreciate it though!

**Comment by Zorro666** [comment]
I think for resize events you need to add a delegate to the window. Something like this in objective-c

```
@implementation myWindowDelegate

- (void)windowDidResize:(NSNotification *)notification
{
..... YOUR CODE ....
}

@end
```

and associate it with the window

```
id delegate = [myWindowDelegate alloc] init];
[nsWindow setDelegate:delegate];

```

The Cube example in MoltenVk has resize handling code but does not have it hooked up on Mac to the resize event.

Another way I have seen it done is at the start of your render method get the size of the `layer` and detect a resize that way.

**Reply by mbechard** [reply]
Thanks, ya I have resizing 'working' and everything. there is just a bug where the swap chain gets corrupted and I get a 'GPU Soft Fault Error' often. Since the setup of the layers/view in MoltenVK feels a little hand-wavy, I was just hoping for an example that was closer to my usage pattern so I had an A-B comparison I could do to see what I'm doing wrong.
